Testing matrix (#256)
plone/meta: README.md

Reviewer: I skipped review of this file, as I already reviewed it in https://github.com/plone/meta/pull/258. After that PR is merged, then I'll revisit this file.

```

#### `.github/workflows/test-matrix.yml`

Run the distribution test on a combination of Plone and Python versions.

> [!NOTE]

> > See the `test_matrix` option in [`tox`](#toxini) configuration file.

Reviewer: This relative anchor does not exist.

Author: It should, it points to the 

```
#### `tox.ini`
```

and from what I see, the dot gets removed. I will double check though

Author: @stevepiercy at least on GitHub UI if you click in [this link](https://github.com/plone/meta/tree/testing-matrix?tab=readme-ov-file#toxini) it leads you to the `tox.ini` section, and as you can see it removes the dot on the anchor.

We are expecting users to see this on GitHub, right? 🤔 so far this is not being imported in docs.plone.org

Reviewer: Ah, OK, ignore this one. I confirmed GitHub strips `.` from anchors.

> [!TIP]

> > 🍀 `plone.meta` tries to be a bit more environmental friendly.

Reviewer: ```suggestion
> 🍀 `plone.meta` tries to be a bit more environmentally friendly.
```

"""
```

Specify custom Docker images in the `custom_images` key.

> The dictionary keys needs to be python versions

Reviewer: ```suggestion
The dictionary keys needs to be Python versions and the value a Docker image for that Python version.
```

> and the value a docker image for that python version.

Reviewer: Semantic line breaks are evil. Please don't use them.

```suggestion
```

"""
```

`plone.meta` generates a list of Python and Plone version combinations to run the distribution tests.

You can customize that by defining your testing matrix:

```toml
[tox]
test_matrix = { "6.1" = ["3.13", "3.10"], "6.0": ["3.13", "3.9"] }

> ```

Reviewer 2: We should say a bit more explicitly how this works. It's not obvious which one is the Plone version and which one is the Python version unless you recognize the numbers.

Author: Oh, I think I mention it on the upgrade docs but not on the readme indeed.

```

When using `plone.meta` outside of plone core packages, there might be extra version pins, or overrides over the official versions.
To specify a custom constraints file, use the `constraints_files` key.

Generating a custom `constraints.txt` is out of scope for `plone.meta` itself.
There are plenty of tools that can do that though.

> [!WARNING]

> > You need to specify the same Plone versions as on `test_matrix` option

Reviewer: ```suggestion
> You need to specify the same Plone versions as in the `test_matrix` option or the default provided by `plone.meta`.
```

> > (or the default provided by `plone.meta`)

Reviewer: ```suggestion
```
